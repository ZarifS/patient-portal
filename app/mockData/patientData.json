[
    {
        "id": 1,
        "firstName": "John",
        "lastName": "Doe",
        "age": 35,
        "gender": "male",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "in-progress",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 2,
        "firstName": "Jane",
        "lastName": "Doe",
        "age": 28,
        "gender": "female",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 3,
        "firstName": "Bob",
        "lastName": "Smith",
        "age": 50,
        "gender": "male",
        "videoUploadStatus": "in-progress",
        "scoliosisPredictionStatus": "not-started",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 4,
        "firstName": "Alice",
        "lastName": "Johnson",
        "age": 40,
        "gender": "female",
        "videoUploadStatus": "not-started",
        "scoliosisPredictionStatus": "not-started",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 5,
        "firstName": "Charlie",
        "lastName": "Brown",
        "age": 45,
        "gender": "male",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 6,
        "firstName": "Lucy",
        "lastName": "Van Pelt",
        "age": 30,
        "gender": "female",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "in-progress",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 7,
        "firstName": "Linus",
        "lastName": "Van Pelt",
        "age": 33,
        "gender": "male",
        "videoUploadStatus": "in-progress",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 8,
        "firstName": "Sally",
        "lastName": "Brown",
        "age": 25,
        "gender": "female",
        "videoUploadStatus": "not-started",
        "scoliosisPredictionStatus": "not-started",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 9,
        "firstName": "Schroeder",
        "lastName": "Unknown",
        "age": 32,
        "gender": "male",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 10,
        "firstName": "Peppermint",
        "lastName": "Patty",
        "age": 29,
        "gender": "female",
        "videoUploadStatus": "in-progress",
        "scoliosisPredictionStatus": "in-progress",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 11,
        "firstName": "Marcie",
        "lastName": "Unknown",
        "age": 31,
        "gender": "female",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 12,
        "firstName": "Rerun",
        "lastName": "Van Pelt",
        "age": 27,
        "gender": "male",
        "videoUploadStatus": "not-started",
        "scoliosisPredictionStatus": "not-started",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 13,
        "firstName": "Pigpen",
        "lastName": "Unknown",
        "age": 33,
        "gender": "male",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "in-progress",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 14,
        "firstName": "Frieda",
        "lastName": "Unknown",
        "age": 36,
        "gender": "female",
        "videoUploadStatus": "in-progress",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 15,
        "firstName": "Violet",
        "lastName": "Gray",
        "age": 34,
        "gender": "female",
        "videoUploadStatus": "not-started",
        "scoliosisPredictionStatus": "not-started",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 16,
        "firstName": "Shermy",
        "lastName": "Unknown",
        "age": 37,
        "gender": "male",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 17,
        "firstName": "Patty",
        "lastName": "Unknown",
        "age": 38,
        "gender": "female",
        "videoUploadStatus": "in-progress",
        "scoliosisPredictionStatus": "in-progress",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 18,
        "firstName": "Franklin",
        "lastName": "Unknown",
        "age": 40,
        "gender": "male",
        "videoUploadStatus": "not-started",
        "scoliosisPredictionStatus": "not-started",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 19,
        "firstName": "Roy",
        "lastName": "Unknown",
        "age": 42,
        "gender": "male",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 20,
        "firstName": "Woodstock",
        "lastName": "Unknown",
        "age": 35,
        "gender": "male",
        "videoUploadStatus": "in-progress",
        "scoliosisPredictionStatus": "in-progress",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 21,
        "firstName": "Spike",
        "lastName": "Unknown",
        "age": 38,
        "gender": "male",
        "videoUploadStatus": "not-started",
        "scoliosisPredictionStatus": "not-started",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 22,
        "firstName": "Belle",
        "lastName": "Unknown",
        "age": 40,
        "gender": "female",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 23,
        "firstName": "Marbles",
        "lastName": "Unknown",
        "age": 45,
        "gender": "male",
        "videoUploadStatus": "in-progress",
        "scoliosisPredictionStatus": "in-progress",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 24,
        "firstName": "Olarf",
        "lastName": "Unknown",
        "age": 50,
        "gender": "male",
        "videoUploadStatus": "not-started",
        "scoliosisPredictionStatus": "not-started",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 25,
        "firstName": "Andy",
        "lastName": "Unknown",
        "age": 55,
        "gender": "male",
        "videoUploadStatus": "completed",
        "scoliosisPredictionStatus": "completed",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 26,
        "firstName": "Molly",
        "lastName": "Volley",
        "age": 30,
        "gender": "female",
        "videoUploadStatus": "in-progress",
        "scoliosisPredictionStatus": "in-progress",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 27,
        "firstName": "Crybaby",
        "lastName": "Boobie",
        "age": 35,
        "gender": "male",
        "videoUploadStatus": "not-started",
        "scoliosisPredictionStatus": "not-started",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    },
    {
        "id": 28,
        "firstName": "Missy",
        "lastName": "Unknown",
        "age": 50,
        "gender": "female",
        "videoUploadStatus": "in-progress",
        "scoliosisPredictionStatus": "in-progress",
        "pointCloudDataURL": "https://sampleurl.com/dwsample-pcd-1280.pcd"
    }
]