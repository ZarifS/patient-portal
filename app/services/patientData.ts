import PATIENT_DATA from '../mockData/patientData.json';

export interface PatientData {
    id: number,
    firstName: string;
    lastName: string;
    age: number;
    gender: "male" | "female",
    lastUpdated: string;
    videoUploadStatus: "in-progress" | "completed" | "not-started" | "failed";
    scoliosisPredictionStatus: "in-progress" | "completed" | "not-started" | "failed";
    pointCloudDataURL: string;
}

export interface PatientDataResponse {
    data: PatientData[], 
    nextPage: number | null,
    maxPages: number,
}
    
// Mock an asynchronous API call to retrieve patient data, mimicking pagination
export const retrievePatientData = (page = 1, filters = {}) => {
    // Mimic pagination
    let startIndex = (page - 1) * 15;
    let endIndex = startIndex + 15;
    let maxPages = Math.ceil(PATIENT_DATA.length / 15);
    // Slice the data to return only 10 patients at a time
    let data = PATIENT_DATA.slice(startIndex, endIndex) as PatientData[];
    let nextPage = PATIENT_DATA.length > endIndex ? page + 1 : null;
    // Mock an asynchronous API call
    return new Promise<PatientDataResponse>((resolve, reject) => {
        setTimeout(() => {
            resolve({data, nextPage: nextPage, maxPages});
            // reject(new Error('Failed to retrieve patient data as server is down.'))
        }, 500);
    }
    );
}

// Mock an asynchronous API call to retrieve patient data
export const retrieveAllPatientData = () => {
    // Mock an asynchronous API call
    return new Promise<PatientData[]>((resolve, reject) => {
        setTimeout(() => {
            const data = PATIENT_DATA as PatientData[];
            resolve(data);
            // reject(new Error('Failed to retrieve patient data as server is down.'))
        }, 500);
    }
    );
}

// Mock an asynchronous API call to retrieve patient data by id
export const getPatientById = (id: number) => {
    return new Promise<PatientData>((resolve, reject) => {
        return setTimeout(() => {
            const patient = PATIENT_DATA.find(patient => patient.id == id);
            if(!patient) reject('Patient not found, please try again.');
            resolve(patient as PatientData);
        }, 1500);
    });
}