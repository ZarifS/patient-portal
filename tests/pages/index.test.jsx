import React from 'react'
import { render, waitFor, screen } from '@testing-library/react'
import Home, { RenderDataTable } from '../../app/page'
import { retrieveAllPatientData } from '../../app/services/patientData'
import PATIENT_DATA from '../../app/mockData/patientData.json';

const data = PATIENT_DATA;

// Mock the retrieveAllPatientData service as a jest mock function
jest.mock('../../app/services/patientData', () => ({
  retrieveAllPatientData: jest.fn(),
}));

describe('Home component', () => {
  it('should match snapshot', async () => {
    // Mock the retrieveAllPatientData service
    retrieveAllPatientData.mockImplementation(() => Promise.resolve([]));
    // Render the component
    const { asFragment } = render(<Home />);
    // Wait for the component to load
    await waitFor(() => expect(retrieveAllPatientData).toHaveBeenCalled());
    // Check that the component matches the snapshot
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('should display error message when data fetching fails', async () => {
    const errorMessage = 'Failed to fetch data';
    // Mock the retrieveAllPatientData service with an error
    retrieveAllPatientData.mockImplementation(() => Promise.reject(new Error(errorMessage)));
    // Render the component and check that the error message is displayed
    render(<Home />);
    await waitFor(() => expect(screen.getByText(`There was an error trying to retrieve your data: ${errorMessage}`)).toBeInTheDocument());
  });

  it('RenderDataTable component should match snapshot', () => {
    // Render the component with mocked data
    const { asFragment } = render(<RenderDataTable data={data} />);
    // Check that the component matches the snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});