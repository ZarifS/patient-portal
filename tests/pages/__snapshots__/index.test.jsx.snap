// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component RenderDataTable component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <input
      class="bg-gray-100 text-sm p-2.5 w-80"
      placeholder="Search by name"
      type="text"
    />
    <div
      class="flex mb-2 mt-2"
    >
      <select
        class="bg-gray-100 text-gray-900 text-sm p-2.5 mr-2 w-44"
        id="sorting"
      >
        <option
          value="id"
        >
          Sort by ID
        </option>
        <option
          value="firstName"
        >
          Sort by First Name
        </option>
        <option
          value="lastName"
        >
          Sort by Last Name
        </option>
        <option
          value="age"
        >
          Sort by Age
        </option>
      </select>
      <select
        class="bg-gray-100 text-gray-900 text-sm p-2.5 w-40"
        id="filterBy_gender"
      >
        <option
          value=""
        >
          Filter by gender
        </option>
        <option
          value="male"
        >
          Male
        </option>
        <option
          value="female"
        >
          Female
        </option>
      </select>
    </div>
  </div>
  <div
    class="overflow-x-auto"
  >
    <table
      class="whitespace-normal w-full"
    >
      <thead
        class="mb-2"
      >
        <tr>
          <th
            class="pr-2 text-left pb-2"
          >
            ID
          </th>
          <th
            class="pr-2 text-left pb-2"
          >
            First Name
          </th>
          <th
            class="pr-2 text-left pb-2"
          >
            Last Name
          </th>
          <th
            class="pr-2 text-left pb-2"
          >
            Age
          </th>
          <th
            class="pr-2 text-left pb-2"
          >
            Gender
          </th>
          <th
            class="pr-2 text-left pb-2"
          >
            Video Status
          </th>
          <th
            class="pr-2 text-left pb-2"
          >
            Predicition Status
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            1
          </td>
          <td
            class="pr-5"
          >
            John
          </td>
          <td
            class="pr-5"
          >
            Doe
          </td>
          <td
            class="pr-5"
          >
            35
          </td>
          <td
            class="pr-5"
          >
            male
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
          <td
            class="pr-5"
          >
            in-progress
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            2
          </td>
          <td
            class="pr-5"
          >
            Jane
          </td>
          <td
            class="pr-5"
          >
            Doe
          </td>
          <td
            class="pr-5"
          >
            28
          </td>
          <td
            class="pr-5"
          >
            female
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            3
          </td>
          <td
            class="pr-5"
          >
            Bob
          </td>
          <td
            class="pr-5"
          >
            Smith
          </td>
          <td
            class="pr-5"
          >
            50
          </td>
          <td
            class="pr-5"
          >
            male
          </td>
          <td
            class="pr-5"
          >
            in-progress
          </td>
          <td
            class="pr-5"
          >
            not-started
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            4
          </td>
          <td
            class="pr-5"
          >
            Alice
          </td>
          <td
            class="pr-5"
          >
            Johnson
          </td>
          <td
            class="pr-5"
          >
            40
          </td>
          <td
            class="pr-5"
          >
            female
          </td>
          <td
            class="pr-5"
          >
            not-started
          </td>
          <td
            class="pr-5"
          >
            not-started
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            5
          </td>
          <td
            class="pr-5"
          >
            Charlie
          </td>
          <td
            class="pr-5"
          >
            Brown
          </td>
          <td
            class="pr-5"
          >
            45
          </td>
          <td
            class="pr-5"
          >
            male
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            6
          </td>
          <td
            class="pr-5"
          >
            Lucy
          </td>
          <td
            class="pr-5"
          >
            Van Pelt
          </td>
          <td
            class="pr-5"
          >
            30
          </td>
          <td
            class="pr-5"
          >
            female
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
          <td
            class="pr-5"
          >
            in-progress
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            7
          </td>
          <td
            class="pr-5"
          >
            Linus
          </td>
          <td
            class="pr-5"
          >
            Van Pelt
          </td>
          <td
            class="pr-5"
          >
            33
          </td>
          <td
            class="pr-5"
          >
            male
          </td>
          <td
            class="pr-5"
          >
            in-progress
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            8
          </td>
          <td
            class="pr-5"
          >
            Sally
          </td>
          <td
            class="pr-5"
          >
            Brown
          </td>
          <td
            class="pr-5"
          >
            25
          </td>
          <td
            class="pr-5"
          >
            female
          </td>
          <td
            class="pr-5"
          >
            not-started
          </td>
          <td
            class="pr-5"
          >
            not-started
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            9
          </td>
          <td
            class="pr-5"
          >
            Schroeder
          </td>
          <td
            class="pr-5"
          >
            Unknown
          </td>
          <td
            class="pr-5"
          >
            32
          </td>
          <td
            class="pr-5"
          >
            male
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            10
          </td>
          <td
            class="pr-5"
          >
            Peppermint
          </td>
          <td
            class="pr-5"
          >
            Patty
          </td>
          <td
            class="pr-5"
          >
            29
          </td>
          <td
            class="pr-5"
          >
            female
          </td>
          <td
            class="pr-5"
          >
            in-progress
          </td>
          <td
            class="pr-5"
          >
            in-progress
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            11
          </td>
          <td
            class="pr-5"
          >
            Marcie
          </td>
          <td
            class="pr-5"
          >
            Unknown
          </td>
          <td
            class="pr-5"
          >
            31
          </td>
          <td
            class="pr-5"
          >
            female
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            12
          </td>
          <td
            class="pr-5"
          >
            Rerun
          </td>
          <td
            class="pr-5"
          >
            Van Pelt
          </td>
          <td
            class="pr-5"
          >
            27
          </td>
          <td
            class="pr-5"
          >
            male
          </td>
          <td
            class="pr-5"
          >
            not-started
          </td>
          <td
            class="pr-5"
          >
            not-started
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            13
          </td>
          <td
            class="pr-5"
          >
            Pigpen
          </td>
          <td
            class="pr-5"
          >
            Unknown
          </td>
          <td
            class="pr-5"
          >
            33
          </td>
          <td
            class="pr-5"
          >
            male
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
          <td
            class="pr-5"
          >
            in-progress
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            14
          </td>
          <td
            class="pr-5"
          >
            Frieda
          </td>
          <td
            class="pr-5"
          >
            Unknown
          </td>
          <td
            class="pr-5"
          >
            36
          </td>
          <td
            class="pr-5"
          >
            female
          </td>
          <td
            class="pr-5"
          >
            in-progress
          </td>
          <td
            class="pr-5"
          >
            completed
          </td>
        </tr>
        <tr
          class="hover:bg-indigo-100 px-5 h-10 cursor-pointer even:bg-gray-100 capitalize"
        >
          <td
            class="pr-5"
          >
            15
          </td>
          <td
            class="pr-5"
          >
            Violet
          </td>
          <td
            class="pr-5"
          >
            Gray
          </td>
          <td
            class="pr-5"
          >
            34
          </td>
          <td
            class="pr-5"
          >
            female
          </td>
          <td
            class="pr-5"
          >
            not-started
          </td>
          <td
            class="pr-5"
          >
            not-started
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <p
    class="font-bold text-sm mt-4"
  >
    Page: 1/3
  </p>
  <div
    class="flex justify-start my-1"
  >
    <button
      class="h-8 px-4 text-white bg-indigo-700 hover:bg-indigo-800"
    >
      Next
    </button>
  </div>
</DocumentFragment>
`;

exports[`Home component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container md:px-10 px-5 pb-5"
  >
    <div
      class="max-w-2xl mb-5"
    >
      <h3
        class="text-lg font-bold mb-1"
      >
        Current Patients
      </h3>
      <p>
        You can view the list of patients registered to you below. 
        <strong>
          Clicking
        </strong>
         on any patient will allow you to see more information as well as 3D modelling information.
      </p>
    </div>
    <div>
      <div
        class="flex flex-col"
      >
        <input
          class="bg-gray-100 text-sm p-2.5 w-80"
          placeholder="Search by name"
          type="text"
        />
        <div
          class="flex mb-2 mt-2"
        >
          <select
            class="bg-gray-100 text-gray-900 text-sm p-2.5 mr-2 w-44"
            id="sorting"
          >
            <option
              value="id"
            >
              Sort by ID
            </option>
            <option
              value="firstName"
            >
              Sort by First Name
            </option>
            <option
              value="lastName"
            >
              Sort by Last Name
            </option>
            <option
              value="age"
            >
              Sort by Age
            </option>
          </select>
          <select
            class="bg-gray-100 text-gray-900 text-sm p-2.5 w-40"
            id="filterBy_gender"
          >
            <option
              value=""
            >
              Filter by gender
            </option>
            <option
              value="male"
            >
              Male
            </option>
            <option
              value="female"
            >
              Female
            </option>
          </select>
        </div>
      </div>
      <div
        class="overflow-x-auto"
      >
        <table
          class="whitespace-normal w-full"
        >
          <thead
            class="mb-2"
          >
            <tr>
              <th
                class="pr-2 text-left pb-2"
              >
                ID
              </th>
              <th
                class="pr-2 text-left pb-2"
              >
                First Name
              </th>
              <th
                class="pr-2 text-left pb-2"
              >
                Last Name
              </th>
              <th
                class="pr-2 text-left pb-2"
              >
                Age
              </th>
              <th
                class="pr-2 text-left pb-2"
              >
                Gender
              </th>
              <th
                class="pr-2 text-left pb-2"
              >
                Video Status
              </th>
              <th
                class="pr-2 text-left pb-2"
              >
                Predicition Status
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <p
        class="font-bold text-sm mt-4"
      >
        Page: 1/0
      </p>
      <div
        class="flex justify-start my-1"
      />
    </div>
  </div>
</DocumentFragment>
`;
